python_modules/dagster-graphql/dagster_graphql/schema/pipelines/pipeline.py

```diff
@@ -311,6 +333,23 @@ def resolve_events(self, graphene_info, after=-1):
         events = graphene_info.context.instance.logs_after(self.run_id, cursor=after)
         return [from_event_record(event, self._pipeline_run.pipeline_name) for event in events]
 
+    def _get_run_record(self, instance):
+        if not self._run_record:
+            self._run_record = instance.get_run_records(run_ids=[self.runId])[0]
+        return self._run_record
+
+    def resolve_startTime(self, graphene_info):
+        run_record = self._get_run_record(graphene_info.context.instance)
+        if run_record.start_time is None and self._pipeline_run.status in STARTED_STATUSES:
+            return graphene_info.context.instance.get_run_stats(self.runId).start_time
```

Comment:
I wonder if you could avoid the extra `get_run_stats` fetch when we try to resolve both `startTime` and `endTime` in the backcompat case.

Reply:
We probably could some of the time - if the end time has already filled out